src/app/api/user-request/route.ts

```diff
@@ -84,26 +113,130 @@ export const DELETE = withSession(async ({ session }) => {
 
     if (joinChapterRequest == null) {
       return NextResponse.json(
-        UndoChapterRequestResponse.parse({
+        ManageChapterRequestResponse.parse({
           code: "INVALID_REQUEST",
-          message: "User doesn't have any active request"
+          message: "User doesn't have any active request",
+        }),
+        { status: 400 }
+      );
+    }
+
+    const canApprove =
+      session.user.role === "ADMIN" ||
+      (session.user.role === "CHAPTER_LEADER" &&
+        session.user.ChapterID === joinChapterRequest.chapterId) ||
+      session.user.id === targetUID;
+
+    if (!canApprove) {
+      return NextResponse.json(
+        ManageChapterRequestResponse.parse({
+          code: "UNAUTHORIZED_REQUEST",
+          message: "User doesn't have permission to deny request",
         }),
         { status: 400 }
       );
     }
 
     await prisma.userRequest.delete({
       where: {
-        uid: session.user.id
+        uid: targetUID,
+      },
+    });
+
+    return NextResponse.json(
+      ManageChapterRequestResponse.parse({ code: "SUCCESS" })
+    );
+  } catch (e: any) {
+    return NextResponse.json(
+      ManageChapterRequestResponse.parse({ code: "UNKNOWN" }),
+      { status: 500 }
+    );
+  }
+});
+
+export const PATCH = withSession(async ({ req, session }) => {
+  try {
+    const approveChapterReq = ManageChapterRequest.safeParse(await req.json());
+    if (!approveChapterReq.success) {
+      return NextResponse.json(
+        ManageChapterRequestResponse.parse({
+          code: "INVALID_REQUEST",
+          message: "Invalid request body",
+        }),
+        { status: 400 }
+      );
+    }
+
+    const targetUID = approveChapterReq.data.userId;
+    const target = await prisma.user.findFirst({
```

Comment:
Can we potentially collapse the cases here? If we search for `UserRequest` and it does not exist, then `targetUID` must be invalid.